runtime/src/main/java/org/corfudb/runtime/collections/PersistedStreamingMap.java

```diff
@@ -0,0 +1,317 @@
+        } catch (RocksDBException ex) {
+            throw new UnrecoverableCorfuError(ex);
+        } finally {
+            keyPayload.release();
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * Please use {@link StreamingMap#entryStream()}.
+     */
+    @Override
+    public boolean containsValue(@NonNull Object value) {
+        throw new UnsupportedOperationException();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public V get(@NonNull Object key) {
+        final ByteBuf keyPayload = Unpooled.buffer();
+        serializer.serialize(key, keyPayload);
+
+        try {
+            byte[] value = rocksDb.get(
+                    keyPayload.array(), keyPayload.arrayOffset(), keyPayload.readableBytes());
+            if (value == null) {
+                return null;
+            }
+            return (V) serializer.deserialize(Unpooled.wrappedBuffer(value), corfuRuntime);
+        } catch (RocksDBException ex) {
+            throw new UnrecoverableCorfuError(ex);
+        } finally {
+            keyPayload.release();
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public V put(@NonNull K key, @NonNull V value) {
+        final ByteBuf keyPayload = Unpooled.buffer();
+        final ByteBuf valuePayload = Unpooled.buffer();
+        serializer.serialize(key, keyPayload);
+        serializer.serialize(value, valuePayload);
+
+        try {
+            rocksDb.put(
+                    keyPayload.array(), keyPayload.arrayOffset(), keyPayload.readableBytes(),
+                    valuePayload.array(), valuePayload.arrayOffset(), valuePayload.readableBytes());
+        } catch (RocksDBException ex) {
+            throw new UnrecoverableCorfuError(ex);
+        } finally {
+            keyPayload.release();
+            valuePayload.release();
+        }
+
+        dataSetSize.incrementAndGet();
+        return value;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public V remove(@NonNull Object key) {
+        final ByteBuf keyPayload = Unpooled.buffer();
+        serializer.serialize(key, keyPayload);
+        try {
+            V value = get(key);
+            if (value != null) {
+                rocksDb.delete(
+                        keyPayload.array(), keyPayload.arrayOffset(), keyPayload.readableBytes());
+                dataSetSize.decrementAndGet();
+                return value;
+            } else {
+                return null;
+            }
+        } catch (RocksDBException ex) {
+            throw new UnrecoverableCorfuError(ex);
+        } finally {
+            keyPayload.release();
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void putAll(@NonNull Map<? extends K, ? extends V> map) {
+        throw new UnsupportedOperationException();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void clear() {
+        entryStream().map(Entry::getKey).forEach(this::remove);
```

Comment:
You can use `DestroyDB` here

Reply:
According to their documentation, calling `DestroyDB()` on a live DB is an undefined behavior. The correct approach is to close the DB then destroy the DB. However, opening and closing the DB mid-air seems somewhat dangerous to me, since we now also have to deal with all IO exceptions, from which we might not be able to recover.

Reply:
Turns out that we end up recreating the object in case of a reset-type call. So `DestroyDB()` is called implicitly in this case.

Reply:
clear() does not end up recreating new object.